src/oscar/apps/catalogue/abstract_models.py

```diff
@@ -1228,23 +1230,91 @@ class AbstractOption(models.Model):
     BOOLEAN = "boolean"
     DATE = "date"
 
+    SELECT = "select"
+    RADIO = "radio"
+    MULTI_SELECT = "multi_select"
+    CHECKBOX = "checkbox"
+
     TYPE_CHOICES = (
         (TEXT, _("Text")),
         (INTEGER, _("Integer")),
         (BOOLEAN, _("True / False")),
         (FLOAT, _("Float")),
         (DATE, _("Date")),
+        (SELECT, _("Select")),
+        (RADIO, _("Radio")),
+        (MULTI_SELECT, _("Multi select")),
+        (CHECKBOX, _("Checkbox")),
     )
 
+    empty_label = "------"
+    empty_radio_label = _("Skip this option")
+
     name = models.CharField(_("Name"), max_length=128, db_index=True)
     code = AutoSlugField(_("Code"), max_length=128, unique=True, populate_from='name')
     type = models.CharField(_("Type"), max_length=255, default=TEXT, choices=TYPE_CHOICES)
     required = models.BooleanField(_("Is this option required?"), default=False)
+    option_group = models.ForeignKey(
+        "catalogue.AttributeOptionGroup",
+        blank=True,
+        null=True,
+        on_delete=models.CASCADE,
+        related_name="product_options",
+        verbose_name=_("Option Group"),
+        help_text=_('Select an option group if using type "Option" or "Multi Option"'),
+    )
+    help_text = models.CharField(
+        verbose_name=_("Help text"), blank=True, null=True, max_length=255
+    )
+    order = models.IntegerField(
```

Comment:
I'm sorry that would be a lot of hassle on our side because we already use this code and I'd have to introduce another migration that doesn't really change anything. So since your request does not really solve a problem and just introduce limitations, I'd like to keep it like this.

Reply:
OK